packages/client/wallets/aa/src/blockchain/wallets/FireblocksNCWallet.ts

```diff
@@ -34,7 +38,7 @@ export const FireblocksNCWallet = async (
         _deviceId = ncwData.deviceId;
         isNew = false;
     } else {
-        const ncwData = await crossmintService.getOrAssignWallet(userEmail);
+        const ncwData = getNCWDataFromLocalStorage() ?? (await crossmintService.getOrAssignWallet(userEmail));
```

Comment:
then we can use like:
`const localStorageRepository = new LocalStorageRepository()` (above on line 32)

```suggestion
        const ncwData = localStorageRepository.ncwAddress ?? (await crossmintService.getOrAssignWallet(userEmail));
```